app/controllers/assignments_controller.rb

```diff
@@ -38,7 +37,7 @@ def create
 
   def update
     if @assignment.update(assignment_params)
-      if current_user.superadmin?
+      if current_user.superadmin? || current_user.department_manager?
```

Comment:
Maybe you could also do `authorize @assignment` before the `if` and do the redirect in a `rescue NotAuthorizedError`? Not sure if it's really better but I think it would be good to keep the rules in the policy.